apiserver/pkg/embedder/embedder.go

```diff
@@ -248,6 +262,12 @@ func ListEmbedders(ctx context.Context, c dynamic.Interface, input generated.Lis
 
 	totalCount := len(us.Items)
 
+	// if pageSize is -1 which means unlimited pagesize,return all
+	if pageSize == common.UnlimitedPageSize {
+		page = 1
+		pageSize = totalCount
```

Comment:
```
return &generated.PaginatedResult{
		TotalCount:  totalCount,
		HasNextPage: end < totalCount,
		Nodes:       result,
	}, nil
```

maybe in line 298, I guess this total is inaccurate, If keyword is not empty, the results have been filtered, but the total number has not changed. HasNextPage will always be true...

It should be a problem elsewhere.